frontend/app/src/App.tsx

```diff
@@ -865,17 +909,12 @@ export class App extends PureComponent<Props, State> {
       pageScriptHash: newPageScriptHash,
     } = newSessionProto
 
-    // mainPage must be a string as we're guaranteed at this point that
-    // newSessionProto.appPages is nonempty and has a truthy pageName.
-    // Otherwise, we'd either have no main script or a nameless main script,
-    // neither of which can happen.
-    const mainPage = newSessionProto.appPages[0] as AppPage
-    // We're similarly guaranteed that newPageName will be found / truthy
-    // here.
-    const newPageName = newSessionProto.appPages.find(
-      p => p.pageScriptHash === newPageScriptHash
-    )?.pageName as string
-    const viewingMainPage = newPageScriptHash === mainPage.pageScriptHash
+    const newSessionHash = hashString(
+      this.sessionInfo.current.installationId + mainScriptPath
+    )
+
+    this.metricsMgr.setMetadata(this.state.deployedAppMetadata)
+    this.metricsMgr.setAppHash(newSessionHash)
```

Comment:
not sure if we want to move this block of code here since it moves the definition of `newSessionHash` far above its main non-metrics usage on L993

Reply:
Hmmm. The intention is to move this up earlier because the updateReport would need to be used earlier.

Perhaps with the ideal [final output](https://github.com/streamlit/streamlit/blob/feature/mpa-v2-integration2/frontend/app/src/App.tsx#L958), it might look a bit better and make more sense?